client/util/format-date.ts

```diff
@@ -0,0 +1,51 @@
+/**
+ * OPINIONED date formatter to use browser locale by default
+ */
+
+import { format as dateFnsFormat, Locale, parseISO, formatDistance as dateFnsFormatDistance } from 'date-fns'
+import { enUS, ja } from 'date-fns/locale'
+
+const locales = { enUS, ja } as const
+
+type ISO8601String = string
+type DateInput = Date | ISO8601String | number
+
+export interface Environment {
+  getLocale(): Locale
+}
+
+/**
+ * defaultEnvironment: Suggest user environment by browser setting.
+ * In the future may use environment that accepts set locale value from crowi instance.
+ */
+export const defaultEnvironment: Environment = {
+  /**
+   * getLocale: Select date-fns locale.
+   * If there are no apporopriate locale to match user setting, use en-US locale.
+   */
+  getLocale() {
+    // TODO: choose language locale by crowi setting
+    const [language, region] = (navigator.userLanguage || navigator.language || '').split('-')
+    return locales[language + region] || locales[language] || locales.enUS
+  },
+}
+
+function sanitizeDateInputForDateFns(input: DateInput) {
+  return typeof input === 'string' ? parseISO(input) : input
+}
+
+export default function format(date: DateInput, format: string, environment: Environment = defaultEnvironment) {
+  return dateFnsFormat(sanitizeDateInputForDateFns(date), format, { locale: environment.getLocale() })
+}
+
+export function formatToLocaleString(date: DateInput, environment: Environment = defaultEnvironment) {
```

Comment:
This is replacement of moment’s ‘llll’ format.
But date-fns’s ‘PPpp’ format doesn’t equals to moment’s one.

moment: Thu, Sep 4, 1986 8:30 PM
date-fns: May 29, 1453, 12:00:00 AM

It’s breaking change, but I think that it’s not critical problem.